src/main/scala/firrtl/options/StageAnnotations.scala

```diff
@@ -14,6 +17,23 @@ sealed trait StageOption { this: Annotation => }
   */
 trait Unserializable { this: Annotation => }
 
+trait HowToSerialize { this: Annotation =>
+
+  protected def baseFileName: String
+
+  protected def suffix: Option[String]
+
+  def howToSerialize: Option[Stream[Byte]]
```

Comment:
scaladoc missing

Reply:
Testing, can I do a suggestion in response to another comment?
```suggestion
  /** Implementation of custom file emission
    *
    * @example {{{
    *  // In simple cases with just a String, use .getBytes
    *  def howToSerialize: Option[Iterable[Byte]] = Some(myString.getBytes)
    * }}}
    */
  def howToSerialize: Option[Iterable[Byte]]
```
Also, why is this an `Option`? I feel like there's a reason but I don't remember.

Reply:
Should the `getBytes` take an encoding like `UTF-8`?

Reply:
`getBytes` has a form that accepts a `Charset`: https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#getBytes-java.nio.charset.Charset-

Not sure if that's something we need to provide or if deferring to the platform is usually sufficient? I guess it depends on the data.

Reply:
Scaladoc added, switched to `Iterable`. I've also included Jack's example.

I can't recall why I made this an `Option`... You could use to abuse the emission. E.g., you set `howToSerialize` to `None` and then use a non-empty `howToResume` to serialize this to some other, safer, annotation. Not sure if that makes sense. (And to reiterate, I cannot recall the original use case.)